src/lsm/tree.zig

```diff
@@ -294,7 +290,7 @@ pub fn TreeType(comptime TreeTable: type, comptime Storage: type, comptime tree_
                 // This simplifies the mutable table and eliminates compaction for duplicate puts.
             }
 
-            if (!tree.table_immutable.free and tree.table_immutable.snapshot_min < snapshot) {
+            if (!tree.table_immutable.free and tree.table_immutable.snapshot_min <= snapshot) {
```

Comment:
Because `snapshot_min/max` are both inclusive now.

Are there any other places in the source where we might still have the same bug for either of these?

Reply:
> Because `snapshot_min/max` are both inclusive now.

Exactly.

I grepped through and didn't see any others. Usually we compare snapshots via `TableInfo.invisible`, which is already inclusive.